crates/router/src/connector/trustpay.rs

```diff
@@ -113,23 +116,29 @@ impl ConnectorCommon for Trustpay {
             .response
             .parse_struct("trustpay ErrorResponse")
             .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
-        let default_error = trustpay::Errors {
-            code: 0,
-            description: consts::NO_ERROR_CODE.to_string(),
-        };
+        let error_list = response.errors.clone().unwrap_or(vec![]);
+        let option_error_code_message = get_error_code_error_message_based_on_priority(
+            self.clone(),
+            error_list.into_iter().map(|errors| errors.into()).collect(),
+        );
+        let reason = response.errors.map(|errors| {
+            errors
+                .iter()
+                .map(|error| error.description.clone())
+                .collect::<Vec<String>>()
+                .join(" & ")
+        });
         Ok(ErrorResponse {
             status_code: res.status_code,
-            code: response.status.to_string(),
-            message: format!(
-                "{:?}",
-                response
-                    .errors
-                    .as_ref()
-                    .unwrap_or(&vec![])
-                    .first()
-                    .unwrap_or(&default_error)
-            ),
-            reason: response.errors.map(|errors| format!("{:?}", errors)),
+            code: option_error_code_message
+                .clone()
+                .map(|error_code_message| error_code_message.error_code)
+                .unwrap_or(consts::NO_ERROR_CODE.to_string()),
+            // message vary for the same code, so relying on code alone as it is unique
+            message: option_error_code_message
+                .map(|error_code_message| error_code_message.error_code)
```

Comment:
This should be error message right?
```suggestion
                .map(|error_code_message| error_code_message.error_message)
```

Reply:
message is dynamic and varies for the same code, so relying on code alone as it is unique, reference -> https://doc.trustpay.eu/aapi#tpgw-s2s-response-errors